packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+	<div class="background">
+		<Suspense>
+			<tera-model-jupyter-panel
+				asset-id="bcd32c48-1daa-4a52-8ed7-147c1dc4b0b0"
+				:model-configuration-id="modelConfigurationId"
+				:project="props.project"
+				:model="null"
+				:show-kernels="false"
+				:show-chat-thoughts="false"
+				@new-model-saved="addOutputPort"
+				:notebook-session="notebookSession"
+			/>
+		</Suspense>
+	</div>
+</template>
+
+<script setup lang="ts">
+// Proxy to use tera-dataset via a workflow context
+
+import { IProject } from '@/types/Project';
+import { WorkflowNode, WorkflowPortStatus } from '@/types/workflow';
+import TeraModelJupyterPanel from '@/components/model/tera-model-jupyter-panel.vue';
+import { computed, onMounted, ref } from 'vue';
+import { workflowEventBus } from '@/services/workflow';
+import { createNotebookSession, getNotebookSessionById } from '@/services/notebook-session';
+import { v4 as uuidv4 } from 'uuid';
+import { NotebookSession } from '@/types/Types';
+import { cloneDeep } from 'lodash';
+import { getModel, getModelConfigurations } from '@/services/model';
+import { addDefaultConfiguration } from '@/services/model-configurations';
+
+const props = defineProps<{
+	node: WorkflowNode;
+	project: IProject;
+}>();
+const modelConfigurationId = computed(() => {
+	// for now we are only using 1 model configuration for the llm at a time, this can be expanded in the future
+	const modelConfirgurationList = props.node?.inputs
+		.filter((inputNode) => inputNode.status === WorkflowPortStatus.CONNECTED && inputNode.value)
+		.map((inputNode) => inputNode.value![0]);
+
+	return modelConfirgurationList[0];
+});
+
+const notebookSession = ref(<NotebookSession | undefined>undefined);
+
+onMounted(async () => {
+	let notebookSessionId = props.node.state?.notebookSessionId;
+	if (!notebookSessionId) {
+		// create a new notebook session log if it does not exist
+		const response = await createNotebookSession({
+			id: uuidv4(),
+			name: props.node.id,
+			description: '',
+			data: { history: [] },
+			timestamp: new Date().toISOString()
+		});
+		notebookSessionId = response?.id;
+
+		if (notebookSessionId) {
+			// update the node state with the notebook session id
+			const state = cloneDeep(props.node.state);
+			state.notebookSessionId = notebookSessionId;
+			workflowEventBus.emit('update-state', {
```

Comment:
Your state setup is strange - I know that the typing information is lost after passing it through a prop. But if we type it out:

```
const state = cloneDeep(props.node.state) as ModelTransformState; 
```

And we don't even have a `ModelTransformState` here, neither is the notion of `state.notebookSessionId`

Incidentally if you know a good way to enforce types please let me know :)

Reply:
fair, I hadn't used the state type at all and will get into the habit of using it.  Change will be made.  

Im not too sure about enforcing types, will have to be something I can look into

Reply:
i've typed the states, I've also done the some with the dataset transformer